cmd/clusterimageset-updater/main.go

```diff
@@ -223,6 +252,47 @@ func main() {
 	}
 }
 
+func ensureLabelsOnClusterPool(s *json.Serializer, input, output string) error {
+	raw, err := os.ReadFile(input)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %w", input, err)
+	}
+	pool := hivev1.ClusterPool{}
+	_, _, err = s.Decode(raw, nil, &pool)
+	if err != nil {
+		return fmt.Errorf("failed to decode from %s: %w", input, err)
+	}
+	if newPool, modified := ensureLabels(pool); modified {
+		logrus.WithField("namespace", newPool.Namespace).WithField("name", newPool.Name).
+			Info("Adding labels to the cluster pool")
+		buf := new(bytes.Buffer)
+		if err := s.Encode(&newPool, buf); err != nil {
+			return fmt.Errorf("failed to encode the new pool from %s: %w", input, err)
+		}
+		if err := os.WriteFile(output, buf.Bytes(), 0644); err != nil {
+			return fmt.Errorf("failed to write updated file %s: %w", output, err)
+		}
+	}
+	return nil
+}
+
+func ensureLabels(pool hivev1.ClusterPool) (hivev1.ClusterPool, bool) {
+	var modified bool
+	if value, ok := pool.Labels["owner"]; ok {
+		if pool.Spec.Labels == nil {
+			pool.Spec.Labels = map[string]string{}
+		}
+		if pool.Spec.Labels["tp.openshift.io/owner"] != value {
+			modified = true
+			pool.Spec.Labels["tp.openshift.io/owner"] = value
+		}
+	} else {
+		logrus.WithField("namespace", pool.Namespace).WithField("name", pool.Name).
+			Warn("The cluster pool has no owner label")
+	}
```

Comment:
Would it be worth enforcing that this `owner` label is required?

Reply:
I thought about this too and my logic is
- Once this is merged and proven working in our production, I will update https://docs.ci.openshift.org/docs/how-tos/cluster-claim/ to tell that the owner label will bring good impact for debugging which should be enough to convince the owners to do the right thing, hopefully. The log here is just to show which pools do not use it at the moment.
- There is a default value by Hive already.

<img width="1315" alt="Screenshot 2023-03-03 at 4 04 59 PM" src="https://user-images.githubusercontent.com/4013349/222829180-b437a1bf-a13d-469e-9de8-ed174f240f51.png">

Reply:
Even better: It is kind of implicitly enforced already, i.e., if a pool has no owner label, it will never be selected by any CI test.

https://github.com/openshift/ci-tools/blob/ecd2e4e9bb1f043ee926d99b8738b6969696a686/pkg/validation/test.go#L473-L475